cavern/src/main/java/org/opencadc/cavern/FileSystemNodePersistence.java

```diff
@@ -233,8 +246,40 @@ public Node put(Node node) throws NodeNotSupportedException, TransientException
     }
 
     @Override
-    public Node updateProperties(Node node, List<NodeProperty> list) throws TransientException {
-        throw new UnsupportedOperationException();
+    public Node updateProperties(Node node, List<NodeProperty> properties) throws TransientException {
+        // node arg is the current node, properties are the new props
+        try {
+            Path np = NodeUtil.nodeToPath(root, node);
+            for (NodeProperty prop : properties)
+            {
+                NodeProperty cur = node.findProperty(prop.getPropertyURI());
+                if (cur != null) {
+                    if (prop.isMarkedForDeletion()) { // delete
+                        //node.getProperties().remove(cur);
+                        cur.setMarkedForDeletion(true);
+                    } else { // update
+                        cur.setValue(prop.getPropertyValue());
+                    }
+                } else { // add
+                    if (!prop.isMarkedForDeletion()) {
+                        node.getProperties().add(prop);
+                    } // else: ignore delete non-existent prop
+                }
+            }
```

Comment:
What about properties whose values have changed?

Reply:
changed values are handled in the cur != null branch above a(at line 261). I'll admit that the FileSystemNodePersistence probably needs unit tests now as it started to have some logic and  not just wrapping calls to NodeUtil...  either that or I put that login in the wrong place. That is something we could decide when we review the public API(s) (after we get everything to work)